bftengine/src/bftengine/ViewsManager.hpp

```diff
@@ -67,23 +67,51 @@ class ViewsManager {
   ///////////////////////////////////////////////////////////////////////////
 
   // should only be called by the primary of the current active view
-  NewViewMsg* getMyNewViewMsgForCurrentView();
+  NewViewMsg *getMyNewViewMsgForCurrentView();
 
-	vector<ViewChangeMsg*> getViewChangeMsgsForCurrentView();
-	NewViewMsg*	getNewViewMsgForCurrentView();
+  vector<ViewChangeMsg *> getViewChangeMsgsForCurrentView();
+  NewViewMsg *getNewViewMsgForCurrentView();
 
   SeqNum stableLowerBoundWhenEnteredToView() const;
 
   struct PrevViewInfo {
-    PrePrepareMsg* prePrepare;
-    bool hasAllRequests;
-    PrepareFullMsg* prepareFull;
+    PrePrepareMsg *prePrepare = nullptr;
+    PrepareFullMsg *prepareFull = nullptr;
+    bool hasAllRequests = true;
+
+    PrevViewInfo() = default;
+
+    PrevViewInfo(PrePrepareMsg *prePrep, PrepareFullMsg *prepFull,
+                 bool allRequests) : prePrepare(prePrep), prepareFull(prepFull),
+                                     hasAllRequests(allRequests) {}
+
+    bool equals(const PrevViewInfo &other) const {
+      if (other.hasAllRequests != hasAllRequests)
+        return false;
+      if ((other.prePrepare && !prePrepare) ||
+          (!other.prePrepare && prePrepare))
+        return false;
+      if ((other.prepareFull && !prepareFull) ||
+          (!other.prepareFull && prepareFull))
+        return false;
+      bool res1 = prePrepare ? (other.prePrepare->equals(*prePrepare)) : true;
+      bool res2 =
+          prepareFull ? (other.prepareFull->equals(*prepareFull)) : true;
+      return res1 && res2;
+    }
+
+    static uint32_t maxSize() {
+      return (PrePrepareMsg::maxSizeOfPrePrepareMsgInLocalBuffer() +
```

Comment:
I suggest to compute this in the code that deals with persistency

Reply:
I prefer to have it written in object-oriented style. I replaced operator '==' by 'equals' function, so it should not hurt to other functionality.